### ISSUE TITLE
Use of aria-required on a select element trigges an error with unrelated feedback

### ISSUE THREAD

**Issue by rianrietveld** [validator/validator#949]
Adding aria-required as attribute to the select element gives the error warning: “Element select is missing one or more of the following attributes: aria-expanded, role”. 

Which is not helpful to the tester and can cause developers to unnecessarily add a role and extra aria to an element that does not need that.

Is aria-required for a select a special case? Because I could not find any specific data on this.

**Comment by sideshowbarker** [comment]
Please post a snippet showing the exact `select` element that‘s generating the error.

**Comment by stevefaulkner** [comment]
test case https://cdpn.io/stevef/debug/ZEbBmMO

**Comment by rianrietveld** [comment]
@sideshowbarker

For example on:
```
<form id="form5">
  <label for="select5">Select</label>
  <select id="select5" aria-required="true">
    <option value="">Choose</option>
	  <option value="One">One</option>
	  <option value="Two">Two</option>
	  <option value="Three">Three</option>
  </select>
  <button>Submit 5</button>
</form>

```
I created a code pen with different combinations of required, aria-required and aria-invalid on a select, and only adding the aria-requiredtriggers that error.
My complete research is on:
https://rianrietveld.com/2020/04/w3c-validator-feedback-for-aria-required-on-a-select-element/

**Comment by stevefaulkner** [comment]
Seems like that `aria-expanded` is required with a `role=combobox`
```<div role=combobox>```, but for a `select` element which has an implicit role of `combobox`, this is not required as the browser exposes `expanded` state under the hood.

![firefox acc properties view showing collapsed state for select element](https://user-images.githubusercontent.com/835859/79960660-9394c380-847d-11ea-9780-cb92462ac84f.png)

**Comment by sideshowbarker** [comment]
@rianrietveld Thanks much that’s exactly what I needed. And thanks @stevefaulkner also.

@stevefaulkner wrote:
> Seems like that `aria-expanded` is required with a `role=combobox`
> `<div role=combobox>`, but for a `select` element which has an implicit role of `combobox`, this is not required as the browser exposes `expanded` state under the hood.

What spec states that `aria-expanded` isn’t required for an implicit combobox?

I don’t see that stated in https://w3c.github.io/html-aria/#index-aria-combobox nor in https://www.w3.org/TR/wai-aria-1.1/#combobox.

What I do see in https://www.w3.org/TR/wai-aria-1.1/#combobox is that both `aria-controls` and `aria-expanded` are listed as **Required States and Properties** for a combobox, period.

**Comment by rianrietveld** [comment]
I don't understand. Does this mean we have to add a role of combobox and a toggle for aria-expanded on a select? And why only if we add aria-required as attribute?

**Comment by stevefaulkner** [comment]
@sideshowbarker your argument appears to imply that `<select>` should trigger an error for missing aria attributes because it implicit role is `combobox`.

Note that [the spec says](https://w3c.github.io/html-aria/#att-required) 

> Use the `aria-required` attribute on any element that is allowed the `required` attribute in HTML. MUST NOT be set to false if the `required` attribute is set

**Comment by sideshowbarker** [comment]
> I don't understand. Does this mean we have to add a role of combobox and a toggle for aria-expanded on a select? And why only if we add aria-required as attribute?

@rianrietveld Yeah, the checker currently only allows `aria-required` for `select` elements if either has an explicit ARIA role, or else if has no explicit ARIA then, the only if it has the additional `aria-` attributes that the ARIA spec defines as being required for an ARIA combobox.

**Comment by sideshowbarker** [comment]
> @sideshowbarker your argument appears to imply that `<select>` should trigger an error for missing aria attributes because its implicit role is `combobox`.

hmm yeah you’re right — indeed that wouldn’t make sense. And that the checker doesn’t report any errors for a plain `<select>`, so the checker is getting that part right at least.

I’ll look at the checker sources now to try and see exactly what condition your example and the example from @rianrietveld are hitting.

**Comment by sideshowbarker** [comment]
OK, after checking the source I can see now that the cause is in fact a checker bug. The current checker sources don’t actually allow a `select` element to be an implicit combobox.

I’ll fix it shortly and push the fix to https://validator.w3.org/nu/

**Comment by sideshowbarker** [comment]
OK, I made the fix to the sources and pushed it to https://validator.w3.org/nu/

Please re-test and let me know if it’s working as expected now

**Comment by rianrietveld** [comment]
Hey @sideshowbarker, thanks so much for the quick respons and the fix. It works ok now.

**Comment by sideshowbarker** [comment]
> Hey @sideshowbarker, thanks so much for the quick respons and the fix. It works ok now.

🎉Thanks very much for having taken time to report it. It caught a lurking checker bug that I’m sure has confused and frustrated other people already — and would have continued to affect more people if you hadn’t called it out. Nice catch :)
